### ISSUE TITLE
Field::likeIgnoreCase should cast lhs operand just like Field::like and Field::similarTo

### ISSUE THREAD

**Issue by kairlec** [jOOQ/jOOQ#13555]
### Your question:

I noticed that jooq will perform `REQUIRES_CAST_ON_LIKE` checks for `LIKE`, `NOT_LIKE`, `SIMILAR_TO`, `NOT_SIMILAR_TO` to generate the necessary CAST statements, and then there is a support check for LIKE_IGNORE_CASE below, I'm wondering why LIKE_IGNORE_CASE Without the necessity of CAST checking, postgre told me that the types didn't match when I did an ilike on a non-String column, and I had to call `DSL.cast` myself to convert it to String
![1652409245(1)](https://user-images.githubusercontent.com/44259087/168200240-fbacb339-ce3d-44d4-bbe4-092f651cff38.jpg)

### Versions:

- jOOQ: 3.14.15
- Java: 17.0.2
- Database (include vendor): postgre
- OS: windows
- JDBC Driver (include name if inofficial driver): org.postgresql.Driver

**Comment by lukaseder** [comment]
Thanks a lot for your message. Let's take a step back. You're asking a question about some *implementation detail*, when in fact your question is more likely about some *API usage* (unless you've found a bug). In either case, can you please describe what you are doing, and what your expectations were, and why they weren't matched? This would include showing some code you wrote.

**Comment by kairlec** [comment]
> Thanks a lot for your message. Let's take a step back. You're asking a question about some _implementation detail_, when in fact your question is more likely about some _API usage_ (unless you've found a bug). In either case, can you please describe what you are doing, and what your expectations were, and why they weren't matched? This would include showing some code you wrote.

```kotlin
val ctx = DSL.using(SQLDialect.POSTGRES)
println(ctx.select().where(DSL.field("some_field").like("%some_value%")))
println(ctx.select().where(DSL.field("some_field").likeIgnoreCase("some_value")))
```
like condition sql is `cast(some_field as varchar) like '%some_value%'`
but like ignore sql is `some_field ilike 'some_value'`

if the field type is not string in postgre, it will cause `You might need to add explicit type casts`

![image](https://user-images.githubusercontent.com/44259087/168714487-048ef622-225a-42a8-b7d3-9aeb007d6690.png)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13569), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13570)
